opflexagent/test/test_agent_types.py

```diff
@@ -38,7 +39,19 @@ def test_opflex_agent_mode(self):
             'opflexagent.as_metadata_manager.AsMetadataManager')
         metadata_patch = metadata_mgr.start()
         cfg.CONF.set_override('agent_mode', 'opflex', 'OPFLEX')
-        with mock.patch('sys.argv'):
+
+        resources = [
+            mock.patch('os.path.basename'),
```

Comment:
before we didn't have to mock the 'os.path.basename' so its that true we have to do that now for python3 support?

Reply:
For py36 and above, we will need to mock the `os.path.basename` because internally it calls `os.fspath`.
https://docs.python.org/3.6/library/os.html?highlight=fspath#os.fspath